OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -96,6 +119,103 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
         }
     }
 
+    private void fetchDateBannerInfo() {
+        if (canFetchBannerInfo && unit.getType() == BlockType.PROBLEM) {
+            courseDateViewModel = new ViewModelProvider(this, new ViewModelFactory()).get(CourseDateViewModel.class);
+            initObserver();
+            courseDateViewModel.fetchCourseDatesBannerInfo(unit.getCourseId(), true);
+        }
+    }
+
+    private void initInfoBanner(CourseBannerInfoModel courseBannerInfo) {
+        if (courseBannerInfo == null) {
+            infoBanner.setVisibility(View.GONE);
+            return;
+        }
+        TextView textView = (TextView) infoBanner.findViewById(R.id.banner_info);
+        Button button = (Button) infoBanner.findViewById(R.id.btn_shift_dates);
+        String buttonText = "";
+        // Currently we are only handling RESET_DATES case,
+        // TODO UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
+        switch (courseBannerInfo.getDatesBannerInfo().getCourseBannerType()) {
+            case UPGRADE_TO_GRADED:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_graded));
+                break;
+            case UPGRADE_TO_RESET:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_reset));
+                break;
+            case RESET_DATES:
+                textView.setText(getText(R.string.course_dates_banner_reset_date));
+                buttonText = getContextOrThrow().getString(R.string.course_dates_banner_reset_date_button);
+                break;
+            case INFO_BANNER:
+                textView.setText(getText(R.string.course_dates_info_banner));
+                break;
+            case BLANK:
+                infoBanner.setVisibility(View.GONE);
+                break;
+        }
+
+        if (!TextUtils.isEmpty(textView.getText())) {
+            if (!TextUtils.isEmpty(buttonText)) {
+                button.setText(buttonText);
+                button.setVisibility(View.VISIBLE);
+                button.setOnClickListener(v -> {
+                    courseDateViewModel.resetCourseDatesBanner(unit.getCourseId());
+                });
+            }
+            infoBanner.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void initObserver() {
+        courseDateViewModel.getBannerInfo().observe(this, this::initInfoBanner);
+
+        courseDateViewModel.getShowLoader().observe(this, flag ->
+                preloadingListener.setLoadingState(flag ?
+                        PreLoadingListener.State.MAIN_UNIT_LOADING :
+                        PreLoadingListener.State.MAIN_UNIT_LOADED));
+
+        courseDateViewModel.getResetCourseDates().observe(this, resetCourseDates -> {
+            if (resetCourseDates != null) {
+                showShiftDateSnackBar(true);
+                canFetchBannerInfo = false;
```

Comment:
no need to add `canFetchBannerInfo` as we have to refresh the course date banner after shifting the date.

Reply:
removed